src/renderer/store/slices/editor/types.ts

```diff
@@ -65,63 +65,121 @@ const editorGraphicalSchema = z.discriminatedUnion('language', [
   }),
 ])
 
+const cursorPositionSchema = z.object({
+  lineNumber: z.number(),
+  column: z.number(),
+  offset: z.number(),
+})
+
+const scrollPositionSchema = z.object({
+  top: z.number(),
+  left: z.number(),
+})
+
+const fbdPositionSchema = z.object({
+  x: z.number(),
+  y: z.number(),
+  zoom: z.number(),
+})
+
 /** This is a zod schema for the model.
  * It is used to validate the model if needed,
  * in most cases you can use the type inferred from it.
  */
 const editorModelSchema = z.discriminatedUnion('type', [
-  z.object({
-    type: z.literal('available'),
-    meta: z.object({
-      name: z.string(),
+  z
```

Comment:
This is a opinion: maybe is a good idea to just use the correct types for each editor. 
Example: fbdPosition only setted to plc-graphical or something like that.